zebra-chain/src/sapling/shielded_data.rs

```diff
@@ -86,6 +86,16 @@ where
     AnchorV: AnchorVariant + Clone,
 {
     /// The net value of Sapling spend transfers minus output transfers.
+    ///
+    /// [`ShieldedData`] structurally validates this [value balance consensus
+    /// rule](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus):
+    /// "If effectiveVersion = 4 and there are no Spend descriptions or Output
+    /// descriptions, then valueBalanceSapling MUST be 0."
+    ///
+    /// `value_balance` is a field in [`ShieldedData`], which must have at least
+    /// one spend or output in its first field, so it is impossible to have a
+    /// [`ShieldedData`] with no spends, no outputs, and a non-zero value
+    /// balance.
```

Comment:
```suggestion
    /// one spend or output in its `transfers` field, If [`ShieldedData`] is `None`
    /// then the `value_balance` is always serialized as zero.
```
I think this part is not totally clear, not sure if my suggestion makes it better.